mypy/types.py

```diff
@@ -816,9 +818,13 @@ def __init__(
         *,
         line: int = -1,
         column: int = -1,
+        min_len: int = 0,
     ) -> None:
         super().__init__(name, fullname, id, upper_bound, default, line=line, column=column)
         self.tuple_fallback = tuple_fallback
+        # This value is not settable by a user. It is an internal-only thing to support
+        # len()-narrowing of variadic tuples.
+        self.min_len = min_len
```

Comment:
Do we need to support this in subtyping, join etc. operations? If there is no clear use case, no need to bother with these, since this is not directly exposed to users. If we'd use this more generally, we'd need to include this in the textual representation, perhaps, which seems something to avoid, since this is not specified by the PEP, I think.

Reply:
I only handle this in subtyping (which btw induces correct behavior for meet), joins are not handled, but I think joins of TypeVarTuples are rare. In general, this is supposed to be "short lived". I wanted to support basic use case where there is an `assert len(args) > 1` followed by some code that uses e.g. `args[1]`.